lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1057,12 +1057,28 @@ def _on_secret_expired(self, event: SecretExpiredEvent) -> None:
         try:
             csr_str = event.secret.get_content(refresh=True)["csr"]
         except ModelError:
-            logger.error("Failed to get CSR from secret - Skipping renewal")
+            logger.error("Failed to get CSR from secret - Skipping")
             return
         csr = CertificateSigningRequest.from_string(csr_str)
         self._renew_certificate_request(csr)
         event.secret.remove_all_revisions()
 
+    def renew_certificate(self, csr: str) -> None:
```

Comment:
I don't think we should pass the CSR itself here, TLS requirers don't manipulate CSR's anymore but those CertificateRequests objects that contain the CSR attributes.

Reply:
Sort of related to this, on the topic of finding better names for classes, I think we should replace:
- `CertificateRequest` -> `CertificateRequestAttributes`

Reply:
Thanks, that's a good point, it isn't that straightforward how to get the CSR anyway even though it's part of the ProviderCertificate object that the requirer has.
I decided to use `ProviderCertificate` instead for the following reasons:
- There is no nice way to go the `CertificateRequest` --> `CertificateSigningRequest` direction and get the CSR. We'd need to do what is currently done in `_certificate_requested_for_attributes` (go the `CertificateSigningRequest` --> `CertificateRequest` direction and check if there is a match) which seems inefficient to do if there is a better way.
- The requirer already has the `ProviderCertificate` object and that object contains the `CertificateSigningRequest` which we need for the renewal.
- It feels nicer to say "_renew this certificate_" than saying "_renew the certificate that belongs to this csr_", especially that this function doesn't allow changing any attributes.

Let me know what you think.

Reply:
I think `CertificateRequestAttributes` is a nice name and should make things a bit clearer.
I'll go ahead and create a separate PR for that.